### ISSUE TITLE
Training code / details

### ISSUE THREAD

**Issue by sgvaze** [ABaldrati/CLIP4CirDemo#1]
Hi!

Thanks for this interesting work! 

I am trying to replicate your results from CIRR and was wondering if you could release the training code? I have evaluated the pre-trained weights from the demo on the evaluation server and get roughly the same results. However, when training the combiner from scratch, I am only getting about 31.5% (global R@1, test set, CLIPR50x4, only train 'combiner'). I am training with full precision on top of pre-extracted features so I can fit the batch size of 4096.

**Comment by zzwDL** [comment]
> Hi!
> 
> Thanks for this interesting work!
> 
> I am trying to replicate your results from CIRR and was wondering if you could release the training code? I have evaluated the pre-trained weights from the demo on the evaluation server and get roughly the same results. However, when training the combiner from scratch, I am only getting about 31.5% (global R@1, test set, CLIPR50x4, only train 'combiner'). I am training with full precision on top of pre-extracted features so I can fit the batch size of 4096.

Hello, I'm also studying this work, but I can't find the training code and model evaluation code in the project provided by the author. I can only run the app.py view examples. Recently, I tried to reproduce the training part of the code, but I found it very difficult. I want to know how you reproduce the results, and whether you could provide your code to my email(zhaozongwei1119@163.com).Thank you very much!

**Comment by ABaldrati** [comment]
Hi!
I apologize for my late reply!!

We are cleaning up the code and we are planning to release it as soon as possible.
I'll post the link to the training code as soon as it is ready.

Alberto

**Comment by sgvaze** [comment]
Great, thanks!

**Comment by ABaldrati** [comment]
Training Code available at https://github.com/ABaldrati/CLIP4Cir !!

**Comment by sgvaze** [comment]
Great, thanks!!

Quick follow up: What do you expect to be the performance of the fine-tuned CLIP model to be on the CIRR val set (e.g for Recall @ 1 when using the entire val set as a retrieval gallery)? Basically, what is the expected val performance after running clip_fine_tune.py (using image+text combine function)?

(I can move this over to the new repo if that's better)

**Comment by ABaldrati** [comment]
> Great, thanks!!
> 
> Quick follow up: What do you expect to be the performance of the fine-tuned CLIP model to be on the CIRR val set (e.g for Recall @ 1 when using the entire val set as a retrieval gallery)? Basically, what is the expected val performance after running clip_fine_tune.py (using image+text combine function)?
> 
> (I can move this over to the new repo if that's better)

HI!
Thanks for your interest in our work!
We are submitting a paper showing that if you fine-tune both the encoders (using the RN50x4 backbone) you can reach a global R@1 of ~44% using the element-wise sum as combining function.
